### PR TITLE
Redone Data Manager, fixed errors

### PR DESCRIPTION
**Pull request by kevingorman1000**
Fixed lots of errors within several modules. 
Reworked Data Manager so now works from one dynamic PHP file. "*TypeManager.php" files within "oc-admin/dataManagement/" are no longer required.
Fixed install errors
Resolved SQL errors within dataActions.php
Resolved SQL file error
Updated "oc-admin/about.php" so now it shows modules loaded, not an array

### CODE REVIEW

**Comment by kevingorman1000** [comment]
Database Entires: 

Duplicate entry within ncicweapons (nameId - UserId) Needed?
Duplicate Entry within ncicplate (nameId - UserId) Needed?
Typo column within ncicplate (vehSSecondaryolor - vehSecondaryColor)
Table doesn't follow naming convention (veh_model - vehModel)

**Comment by phillf** [comment]
@kevingorman1000, are there any queries for `userid` in either ncic toolkit? I think there might be something. I would leave `userid` until it is proven it isn't required.

I am pretty sure that column is written to when ever a record is saved that way all weapons or vehicles can be pulled where a given `userid` is 1 for example.

**Comment by kevingorman1000** [comment]
From the looks of it, it links up to the users table. Which, as you stated, is correct. Go ahead and keep that table. No other NCIC table seems to have it apart from plates and weapons, should warnings, warrants etc not contain that column as well ?

File: oc-admin/about.php [hunk 1]
```diff
@@ -11,143 +11,132 @@
...
-	<?php require_once ( ABSPATH . "/" . OCTHEMES ."/". THEME ."/includes/footer.inc.php"); ?>
-	<?php require_once( ABSPATH . OCTHEMEINC ."/admin/topbarNav.inc.php" ); ?>
-	<?php require_once( ABSPATH . OCTHEMEINC ."/scripts.inc.php" ); ?>
-	</body>	
+					<div class="card">
+						<div class="card-header">
+							<em class="fa fa-align-justify"></em> <?php echo lang_key("ABOUT_OPENCAD"); ?>
+						</div>
+						<div class="card-body">
+							<p>OpenCAD is an open source project licensed under GNU GPL v3. The original code and concept by <a rel="noopener" href="https://github.com/ossified" title="a link to the original developer's GitHub.">Shane Gill</a>. This project is maintained by Overt Source.</p>
```
Comment (ItsAGeekThing) [inline oc-admin/about.php]: We can probably drop the the reference to Overt Source. I'll get clarification on this with @phillf.

```suggestion
							<p>OpenCAD is an open source project licensed under GNU GPL v3. The original code and concept by <a rel="noopener" href="https://github.com/ossified" title="a link to the original developer's GitHub.">Shane Gill</a>.</p>
```

File: oc-api/config/functions.inc.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        echo UnauthorizedApiKey();
+        exit();
+    } else{
+        return true;
+    }
+}
+
+function UnauthorizedApiKey(){
+    
+    http_response_code(201);
```
Comment (ItsAGeekThing) [inline oc-api/config/functions.inc.php]: Set HTTP response status code to "401 Unauthorized".

```suggestion
    http_response_code(401);
```

File: oc-config.sample.php [hunk 3]
```diff
@@ -780,72 +892,72 @@
...
+  * If 'true' OpenCAD will ignore check for the presence of the oc-install directory,
+  * Otherwise, if 'false' then OpenCAD will check for the presence of the oc-install directory.
+  * 
+  * ONLY USE IN DEVELOPMENT ENVIRONMENTS. DO NOT ENABLE IN PRODUCTION
+  *
+  * DEFAULT VALUE: false
+	*
+  * @since OpenCAD OpenCAD version 0.3.1
+  **/
+  define('OC_DEVELOP', true);
```
Comment (ItsAGeekThing) [inline oc-config.sample.php]:
```suggestion
  define('OC_DEVELOP', false);
```

File: oc-settings.php [hunk 4]
```diff
@@ -24,6 +24,7 @@
 define( 'OCTHEMEJS', 'oc-content/themes/'. THEME .'/js' );
 define( 'APIDATAACTIONS', '<form action="'.BASE_URL.'/oc-includes/dataActions.php" method="post">');
 define( 'ERRORREDIRECT', '\'Location: '.BASE_URL.'/oc-content/plugins/error/index.php\'');
+define( 'APPLATESTVERSIONURL', "https://raw.githubusercontent.com/kevingorman1000/OpenCad-Versions-TEMP/main/version.json");
```
Comment (ItsAGeekThing) [inline oc-settings.php:27]: Perhaps this is something we could pull from the GitHub API or maybe even use a CloudFlare worker to create a API endpoint.

https://docs.github.com/en/rest/git/tags#get-a-tag

Comment (kevingorman1000) [reply oc-settings.php:27]: This was intended to be a test feature for now as a "proof of concept". Obviously have to have discussions on the best way to implement it, if we implement it.

Comment (ItsAGeekThing) [reply oc-settings.php:27]: Absolutely.

We can open a separate issue for this to discuss it's (proposed) implementation later on. I'll go ahead and mark this resolved. 👍

File: oc-api/config/functions.inc.php (hunk 2 above)
Comment (kevingorman1000) [reply oc-api/config/functions.inc.php]: I forgot to reply to all of these comments. This has already been resolved, apparently I haven't pushed the updated version? Once I've worked on what I'm doing now, I'll push it with the resolved response code.

**Review by phillf — CHANGES_REQUESTED** [review]
Kevin, you need to pull changes from the upstream before this can be merged. Your README.md is out of date.

**Comment by kevingorman1000** [comment]
Readme has been updated, everything seems to be updated and correct. Let me know if any issues.

**Review by phillf — APPROVED** [review]

**Review by phillf — APPROVED** [review]

**Pull request merged**